server/nodeset/upload-deposit-data.go

```diff
@@ -68,79 +69,109 @@ func (c *nodesetUploadDepositDataContext) PrepareData(data *swapi.NodesetUploadD
 		return types.ResponseStatus_WalletNotReady, err
 	}
 
-	// Get the list of registered validators
-	registeredPubkeyMap := map[beacon.ValidatorPubkey]bool{}
-	pubkeyStatusResponse, err := nc.GetRegisteredValidators(ctx)
+	// Fetch private keys and derive public keys
+	privateKeys, err := w.GetAllPrivateKeys()
 	if err != nil {
-		return types.ResponseStatus_Error, fmt.Errorf("error getting registered validators: %w", err)
+		return types.ResponseStatus_Error, fmt.Errorf("error getting private keys: %w", err)
 	}
-	registeredPubkeys := []beacon.ValidatorPubkey{}
-	for _, pubkeyStatus := range pubkeyStatusResponse {
-		registeredPubkeys = append(registeredPubkeys, pubkeyStatus.Pubkey)
+	publicKeys, err := w.DerivePubKeys(privateKeys)
```

Comment:
Where is `DerivePubKeys()` from? Is that something you added to the daemon that hasn't been PR'd yet?

Reply:
This was always here. It's a function in wallet.go:

https://github.com/nodeset-org/hyperdrive-stakewise/blob/dev/common/wallet.go#L139

Reply:
Got it, thank you!